### PR TITLE
feat(admin): page-number pagination for question list

### LINKED ISSUE #461: feat: cursor-based pagination for admin question list
## Context
The admin questions list is hard-limited to 100 rows with no way to see the rest. PR #460 added a `hasMore` flag and `QUESTION_LIMIT` constant as building blocks, but without actual pagination the user just sees "(limit reached)" and has to narrow filters.

## Root cause
The original design never included pagination — `.limit(100)` was a performance guard without an escape hatch.

## Proposed solution
Cursor-based pagination using `created_at` as the cursor:
- Add `cursor` param to `parseFilters` / `getQuestionsList`
- "Load more" button in `QuestionsPageShell` when `hasMore` is true
- Use `.lt('created_at', cursor)` for the next page
- `QUESTION_LIMIT` and `hasMore` from PR #460 are the foundation

## Building blocks already in place
- `QUESTION_LIMIT = 100` exported constant
- `hasMore: boolean` in `QuestionsListResult`
- `QuestionsPageShell` accepts `hasMore` prop

## Source
Identified during PR #460 review — the `hasMore` flag without pagination was flagged as a cosmetic fix, not a real solution.

### PR DESCRIPTION
**Pull request by okpilot**
## Summary
- Replace hard `LIMIT+1` overfetch with Supabase offset pagination (`.range()` + `count:'exact'`), 25 questions per page
- Add page-number UI with Prev/1/2/3.../Next buttons and "Showing X-Y of Z questions" display
- Server-side redirect for out-of-range pages (e.g. `?page=999` redirects to last valid page)
- Filter changes reset pagination to page 1
- Add spec-workflow MCP to `.mcp.json` (was missing since #462)

Closes #461

## Test plan
- [x] 183 unit tests pass (37 new/updated across 3 test files)
- [x] `pnpm check-types` clean
- [x] Biome lint/format clean
- [x] Security auditor approved push
- [x] Manual: 66 seeded questions, 3 pages render correctly
- [x] Manual: filter change resets to page 1
- [x] Manual: `?page=999` redirects to last valid page
- [ ] CI: all checks green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Page-based pagination for admin questions with Prev/Next, page buttons, and fixed page size (25).  

* **Improvements**
  * Exact totals shown in “Showing X–Y of Z questions”; filter changes reset pagination; out-of-range pages redirect to last valid page; page parsing defaults/truncates to 1; navigation preserves other filters; improved search escaping/behavior.  

* **Documentation**
  * Added design, requirements, tasks, workflow, and pre-push quality checklist.  

* **Tests**
  * Extensive unit tests for parsing, query/count behavior, and pagination UI.

### CODE REVIEW

File: .spec-workflow/specs/admin-questions-pagination/design.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+### Unit Testing
+- `queries.test.ts`: Mock chain gains `.range()`, `then` returns `{ data, count, error }`. Test offset calculation, totalCount, edge cases.
+- `page.test.ts`: Test `page` param parsing (valid, invalid, missing, zero, negative, float).
+- `pagination-bar.test.tsx`: Prev/Next disabled states, page count, active page highlighting.
+
+### Manual Testing
+- Visit with 100+ questions, navigate pages
+- Change filters mid-pagination — verify reset to page 1
+- Visit `?page=999` — shows empty state
+- Visit `?page=-1` or `?page=abc` — defaults to page 1
```
[the remark this answers is not in the document]
Comment (okpilot) [reply .spec-workflow/specs/admin-questions-pagination/design.md]: Fixed in 3f4f74a. Updated to 'redirects to last valid page' — matches the `redirect()` call in `questions-content.tsx:31`.

File: .spec-workflow/specs/admin-questions-pagination/requirements.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+**User Story:** As an admin, I want to browse questions page by page, so that I can access any question in the bank regardless of position.
+
+#### Acceptance Criteria
+
+1. WHEN admin visits `/app/admin/questions` THEN system SHALL display page 1 (first 25 questions) ordered by `created_at DESC`
+2. WHEN admin clicks a page number (e.g. page 3) THEN system SHALL navigate to `?page=3` and display questions 51-75
+3. WHEN admin is on page 1 THEN Prev button SHALL be disabled
+4. WHEN admin is on the last page THEN Next button SHALL be disabled
+5. WHEN `page` param is invalid (negative, zero, non-integer, non-numeric) THEN system SHALL default to page 1
```
[the remark this answers is not in the document]
Comment (okpilot) [reply .spec-workflow/specs/admin-questions-pagination/requirements.md]: Fixed in 3f4f74a. Aligned R1.5 with shipped behavior: `Number.parseInt('2.7', 10)` returns `2` (truncation), which the test at `page.test.ts:105-107` already asserts. Updated spec to distinguish float truncation from invalid-input defaults.

File: apps/web/app/app/admin/questions/_components/questions-content.test.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+const mockGetQuestionsList = vi.hoisted(() => vi.fn())
+const mockGetSyllabusTree = vi.hoisted(() => vi.fn())
+const mockRedirect = vi.hoisted(() => vi.fn())
+
+// ---- Module mocks -----------------------------------------------------------
+
+vi.mock('../queries', () => ({
+  getQuestionsList: mockGetQuestionsList,
+  PAGE_SIZE: 25, // SYNC WITH queries.ts PAGE_SIZE if value changes
+}))
```
[the remark this answers is not in the document]
Comment (okpilot) [reply apps/web/app/app/admin/questions/_components/questions-content.test.tsx]: Acknowledged — we added the sync comment in 9749143 for exactly this reason (`// SYNC WITH queries.ts PAGE_SIZE if value changes`). Using `importOriginal` would add complexity to the mock for a constant that's unlikely to change. The comment makes the coupling visible so a future change triggers a conscious test update.
